Helper/Order.php

```diff
@@ -574,23 +574,24 @@ private function getQuotePaymentInfoInstance($quote)
     }
 
     /**
-     * Delete redundant immutable quotes.
+     * Delete redundant immutable quotes and redundant parent (session) quote
+     * Only the converted immutable quote is preserved
      *
-     * @param Quote $quote
+     * @param Quote $convertedQuote  The non-session quote that was converted to an order
      */
-    private function deleteRedundantQuotes($quote)
+    private function deleteRedundantQuotes($convertedQuote)
     {
         $connection = $this->resourceConnection->getConnection();
 
         // get table name with prefix
-        $tableName = $this->resourceConnection->getTableName('quote');
+        $fromQuoteTable = $this->resourceConnection->getTableName('quote');
 
-        $bind = [
-            'bolt_parent_quote_id = ?' => $quote->getBoltParentQuoteId(),
-            'entity_id != ?' => $quote->getBoltParentQuoteId()
```

Comment:
@danac-gs Here is the significant change #1.  
All cloned quotes are deleted, including the parent quote, except for the cloned quote that the order is based from.

Reply:
This does'n work, at least not with the latest changes. The $quote that is passed to the method (renamed to $convertedQuote) is actually the parent / session quote, with BoltParentQuoteId == Id, so the parent stays, all clones get deleted.